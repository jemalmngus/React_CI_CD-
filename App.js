import React, { Component } from 'react';

import { render } from 'react-dom';
class App extends Component{

  
    render() {

       const footer = {
   position:"fixed",
   left:"0",
   bottom:"0",
   width:"100%",
   backgroundColor:"#111",
   color:"white",
   textAlign:"center"
   
};
  
      return (
     <div>
<ul>
  <li><a href={'#'}>Home</a></li>
  <li><a href={'#'}>News</a></li>
  <li><a href={'#'}>Contact</a></li>
  <li style={{float:"right"}}><a style={{backgroundColor: "#4CAF50"}} href={'#'}>About</a></li>
  
</ul>
<h2>Hello Jemal</h2>
<p>This react demo </p>

<footer style={footer}>
  <p>Footer</p>
</footer>
</div>

         
      )
    }
  }
export default App;
