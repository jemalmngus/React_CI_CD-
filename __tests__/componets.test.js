import React from "react";
import { create } from "react-test-renderer";
import {sum, multiply} from '../maths.js';
import ReactDOM from 'react-dom';
import App from '../App.js';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('renders a snapshot', () => {
    const tree = renderer.create(<App/>).toJSON();
    expect(tree).toMatchSnapshot();
  });