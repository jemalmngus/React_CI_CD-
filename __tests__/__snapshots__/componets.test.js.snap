// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <ul>
    <li>
      <a
        href="#"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        News
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Contact
      </a>
    </li>
    <li
      style={
        Object {
          "float": "right",
        }
      }
    >
      <a
        href="#"
        style={
          Object {
            "backgroundColor": "#4CAF50",
          }
        }
      >
        About
      </a>
    </li>
  </ul>
  <h2>
    Hello Jemal
  </h2>
  <p>
    This react demo 
  </p>
  <footer
    style={
      Object {
        "backgroundColor": "#111",
        "bottom": "0",
        "color": "white",
        "left": "0",
        "position": "fixed",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <p>
      Footer
    </p>
  </footer>
</div>
`;
